### PR TITLE
feat: add ens name and avatar support

### PR DESCRIPTION
**Pull request by bigint**
image

cc @laktek

### CODE REVIEW

File: src/hooks/useENS.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ethers } from "ethers";
+import { useEffect, useState } from "react";
+
+const useENS = (address: string | null | undefined) => {
+  const [ensName, setENSName] = useState<string | null>();
+
+  useEffect(() => {
+    const resolveENS = async () => {
+      if (address && ethers.utils.isAddress(address)) {
+        const provider = ethers.providers.getDefaultProvider();
```
Comment (laktek) [inline src/hooks/useENS.ts:10]: Can you use `useWeb3React("mainnet")` hook instead? Check the `src/hooks/useAddresses.ts` for how I'm resolving the ENS name.

**Comment by laktek** [comment]
Really appreciate the PR! I left a minor suggestion for refactoring. Let me know when you update it, I'll merge the PR.

**Review by laktek — COMMENTED** [review]
Oops..missed to submit the review.

File: src/hooks/useENS.ts (hunk 1 above)
Comment (bigint) [reply src/hooks/useENS.ts:10]: Oops! Unfortunately `@web3-react/core` don't support ens resolver 😢

Comment (laktek) [reply src/hooks/useENS.ts:10]: ok, let's merge this PR as it is and then will do a refactor. Thanks again for the contribution ❤️

**Pull request merged**